### PR TITLE
*: deprecate alter-primary-key configuration

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

This PR removes the config `alter-primary-key`. To alter the primary key, the users should use the [`NONCLUSTERED` syntax](https://github.com/pingcap/tidb/pull/22409) instead to create a table.

### What is changed and how it works?

What's Changed: Omitted.

How it Works: Omitted.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`!

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility

### Release note 

- The configuration `alter-primary-key` is deprecated. Use `NONCLUSTERED` instead.

### CODE REVIEW

**Comment by tangenta** [comment]
Blocked by https://github.com/pingcap/br/pull/853.

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1530,6 +1506,28 @@ func isSingleIntPK(constr *ast.Constraint, lastCol *model.ColumnInfo) bool {
...
+func ShouldBuildClusteredIndex(ctx sessionctx.Context, opt *ast.IndexOption) bool {
+	if opt == nil || opt.PrimaryKeyTp == model.PrimaryKeyTypeDefault {
+		return ctx.GetSessionVars().EnableClusteredIndex
+	}
+	switch opt.PrimaryKeyTp {
+	case model.PrimaryKeyTypeClustered:
+		hasBinlog := ctx.GetSessionVars().BinlogClient != nil
+		if hasBinlog {
+			errMsg := "cannot build clustered index table because the binlog is ON"
+			ctx.GetSessionVars().StmtCtx.AppendWarning(errors.Errorf(errMsg))
```
Comment (jackysp) [inline ddl/ddl_api.go:1519]: Why not raise an error instead?

Comment (tangenta) [reply ddl/ddl_api.go:1519]: @coocood PTAL

Comment (jackysp) [reply ddl/ddl_api.go:1519]: ok, got it.

File: cmd/explaintest/r/clustered_index.result [hunk 2]
```diff
@@ -3,34 +3,32 @@ create database with_cluster_index;
...
-create table tbl_3 ( col_15 tinyint default -91 not null , col_16 bit(61) default 990141831018971350 not null , col_17 double(244,22) default 3985 not null , col_18 binary(32) default 'kxMlWqvpxXNBlxoU' , col_19 text(401) , primary key idx_10 ( col_18,col_19(4) ) , key idx_11 ( col_17,col_18,col_19(2),col_15,col_16 ) , unique key idx_12 ( col_17 ) )  ;
-create table tbl_4 ( col_20 double(230,16) default 8.49 not null , col_21 int unsigned not null , col_22 enum('Alice','Bob','Charlie','David') not null , col_23 float default 3066.13040283622 , col_24 datetime default '1980-10-27' not null , primary key idx_13 ( col_22,col_24 ) , key idx_14 ( col_23,col_20 ) , key idx_15 ( col_24 ) , key idx_16 ( col_20 ) , unique key idx_17 ( col_24 ) , key idx_18 ( col_21 ) )  ;
+create table tbl_0 ( col_0 decimal not null , col_1 blob(207) , col_2 text , col_3 datetime default '1986-07-01' , col_4 bigint unsigned default 1504335725690712365 , primary key idx_0 ( col_3,col_2(1),col_1(6) ) , key idx_1 ( col_3 ) clustered, unique key idx_2 ( col_3 ) , unique key idx_3 ( col_0 ) , key idx_4 ( col_1(1),col_2(1) ) , key idx_5 ( col_2(1) ) )  ;
```
Comment (lysu) [inline cmd/explaintest/r/clustered_index.result:6]: `clustered` need behind `primary key`

**Comment by lysu** [comment]
LGTM for 42e53ae

File: ddl/serial_test.go [hunk 3]
```diff
@@ -53,6 +53,9 @@ import (
 // Make it serial because config is modified in test cases.
 var _ = SerialSuites(&testSerialSuite{})
 
+// TODO(tangenta): Move all the parallel tests out of this file.
+var _ = Suite(&testIntegrationSuite9{&testIntegrationSuite{}})
```
Comment (tisonkun) [inline ddl/serial_test.go:57]: Shall we resolve this TODO now? I think it's in the way of #26022 and we'd really need to split all these huge test files under `ddl` package.

**Pull request merged**